networks/3a_validate_array_vs_rnaseq_modules.py

```diff
@@ -0,0 +1,217 @@
+    pao1_rnaseq_clustering = DBSCAN(eps=density_threshold).fit(
+        pao1_rnaseq_corr_log_spell
+    )
+
+# Clustering using hierarchal clustering
+if cluster_method == "hierarchal":
+    pao1_array_clustering = AgglomerativeClustering(
+        n_clusters=None, distance_threshold=hier_threshold, linkage=link_dist
+    ).fit(ppao1_array_corr_log_spell)
+    pao1_rnaseq_clustering = AgglomerativeClustering(
+        n_clusters=None, distance_threshold=hier_threshold, linkage=link_dist
+    ).fit(pao1_rnaseq_corr_log_spell)
+
+# Clustering using affinity propogation
+if cluster_method == "affinity":
+    pao1_array_clustering = AffinityPropagation(random_state=0).fit(
+        pao1_array_corr_log_spell
+    )
+    pao1_rnaseq_clustering = AffinityPropagation(
+        random_state=1, damping=affinity_damping
+    ).fit(pao1_rnaseq_corr_log_spell)
+
+# +
+# Get module membership for a single threshold
+# Format and save output to have columns: gene_id | group_id
+pao1_array_membership_df = pd.DataFrame(
+    data={"module id": pao1_array_clustering.labels_},
+    index=pao1_array_corr_log_spell.index,
+)
+
+pao1_array_membership_df["module id"].value_counts()
+# -
+
+pao1_array_membership_df.head()
+
+# +
+pao1_rnaseq_membership_df = pd.DataFrame(
+    data={"module id": pao1_rnaseq_clustering.labels_},
+    index=pao1_rnaseq_corr_log_spell.index,
+)
+
+pao1_rnaseq_membership_df["module id"].value_counts()
+# -
+
+pao1_rnaseq_membership_df.head()
+
+# ## Compare composition of modules
+#
+# For a given array module, are the genes within 1 module in the RNA-seq compendium?
+
+# +
+array_module_mapping = {}
+for grp_id, grp in pao1_array_membership_df.groupby("module id"):
+    grp_mapped = pao1_rnaseq_membership_df.loc[grp.index]
+    array_module_mapping[grp_id] = list(grp_mapped["module id"].unique())
+
+array_module_mapping
+
+# +
+# Are there any array modules that map to a single rnaseq module?
+consistent_array_modules = []
+for array_module_id, list_rnaseq_module_ids in array_module_mapping.items():
+    if len(list_rnaseq_module_ids) == 1:
+        print(array_module_id, list_rnaseq_module_ids)
+        consistent_array_modules.append(array_module_id)
+
+print(len(consistent_array_modules) / len(array_module_mapping))
+
+# +
+rnaseq_module_mapping = {}
+for grp_id, grp in pao1_rnaseq_membership_df.groupby("module id"):
+    grp_mapped = pao1_array_membership_df.loc[grp.index]
+    rnaseq_module_mapping[grp_id] = list(grp_mapped["module id"].unique())
+
+rnaseq_module_mapping
+
+# +
+# Are there any rnaseq modules that map to a single array module?
+consistent_rnaseq_modules = []
+for rnaseq_module_id, list_array_module_ids in rnaseq_module_mapping.items():
+    if len(list_array_module_ids) == 1:
+        print(rnaseq_module_id, list_array_module_ids)
+        consistent_rnaseq_modules.append(rnaseq_module_id)
+
+print(len(consistent_rnaseq_modules) / len(rnaseq_module_mapping))
```

Comment:
If I'm not missing something, I think you could use more standard ways to compute the similarity between two clustering partitions, like the adjusted Rand index (ARI, it's available in sklearn). If each rna-seq cluster map to a single array cluster, then you get an ARI=1.0
I know that here you are doing it manually to have a percentage/more interpretable measure, but I wonder what's the ARI value of these two data partitions. Also, you have [other clustering measures](https://scikit-learn.org/stable/modules/classes.html#sklearn-metrics-metrics) (under "Clustering metrics") that let you know, for instance, if clusters in one partition are "contained" in other clusters, what could be interesting for your question I think (like the homogeneity/completeness scores).

Reply:
I mention this because it could happen that there is not a perfect match between clusters in both data types, but maybe in general gene pairs are clustered in the same clusters. You will miss this if you don't use a pair-counting-based measure like ARI.

Reply:
Very good point. I'll look into this.